packages/proxy/lib/http/util/cookies.ts

```diff
@@ -3,13 +3,49 @@ import type Debug from 'debug'
 import { URL } from 'url'
 import { cors } from '@packages/network'
 import { AutomationCookie, Cookie, CookieJar, toughCookieToAutomationCookie } from '@packages/server/lib/util/cookies'
+import { calculateSiteContext } from './top-simulation'
+import type { RequestCredentialLevel, RequestResourceType } from '../../types'
 
 interface RequestDetails {
   url: string
   isAUTFrame: boolean
   needsCrossOriginHandling: boolean
 }
 
+export const shouldAttachAndSetCookies = (requestUrl: string, AUTUrl: string | undefined, resourceType?: RequestResourceType, credentialLevel?: RequestCredentialLevel): boolean => {
+  if (!AUTUrl) return false
+
+  const siteContext = calculateSiteContext(requestUrl, AUTUrl)
+
+  switch (resourceType) {
+    case 'fetch':
+      // never attach cookies regardless of siteContext if omit is optioned
+      if (credentialLevel === 'omit') {
+        return false
+      }
+
+      // if the siteContext is same-origin and the credential status is undefined, same-origin, or include, attach cookies.
+      // Otherwise, if the credentials are set to include, attach cookies
+      if (siteContext === 'same-origin' || credentialLevel === 'include') {
+        return true
+      }
+
+      return false
+    case 'xhr':
+      // if context is same-origin regardless of credential status, attach cookies
+      // otherwise, if withCredentials is set to true, attach cookies
+
+      if (siteContext === 'same-origin' || credentialLevel) {
```

Comment:
based on comment?
```suggestion
      if (siteContext === 'same-origin' {
```

Reply:
I think the comment here isn't very good and I updated it [here](https://github.com/cypress-io/cypress/pull/23880/files#diff-8bcc53731d8dc8219a282445e0ae5ae47e2884d0fb28ead3eef2b488819d1bf0R51). If the site-context is same origin, or the request is sent with credentials, we should attach/set cookies